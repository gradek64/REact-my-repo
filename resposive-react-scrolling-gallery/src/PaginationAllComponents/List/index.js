import React from 'react';
import PropTypes from 'prop-types';

// custom
import Card from '../Item/Card';

/* eslint-disable no-underscore-dangle */
const List = ({ list }) => (
  <div className="paginated-list">
    <ul className="cards-slider-inner">
      {
        list.map((item, key) => <Card key={item._id} {...item} />)
      }
    </ul>
  </div>
);

List.propTypes = {
  list: PropTypes.array.isRequired,
};

export default List;
