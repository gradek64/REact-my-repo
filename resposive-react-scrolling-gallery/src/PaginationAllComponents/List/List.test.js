import React from 'react';
import { shallow } from 'enzyme';
import List from './index';


// no point testing it just it needs to have an array structure
const list = [
  {
    _id: '593e9297e17df20c4a237d42',
    index: 0,
    price: 937180,
    picture: 'images/tn_property01.jpg',
    city: 'Singer',
    address: '914 Argyle Road Alley',
    latitude: -33.944576,
    longitude: 151.25584,
    bedrooms: 2,
    bathrooms: 2,
    carSpaces: 2,
  },
  {
    _id: '593e9297ec4cca9c56bf61af',
    index: 1,
    price: 703725,
    picture: 'images/tn_property02.jpg',
    city: 'Machias',
    address: '255 Raleigh Place',
    latitude: -33.944471,
    longitude: 151.2541,
    bedrooms: 2,
    bathrooms: 1,
    carSpaces: 0,
  }];

test('renders correctly', () => {
  const wrapper = shallow(<List list={list} />);
  expect(wrapper).toMatchSnapshot();
});
