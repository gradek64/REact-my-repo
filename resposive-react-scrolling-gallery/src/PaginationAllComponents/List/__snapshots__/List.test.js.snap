// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="paginated-list"
>
  <ul
    className="cards-slider-inner"
  >
    <Card
      _id="593e9297e17df20c4a237d42"
      address="914 Argyle Road Alley"
      bathrooms={2}
      bedrooms={2}
      carSpaces={2}
      city="Singer"
      index={0}
      key="593e9297e17df20c4a237d42"
      latitude={-33.944576}
      longitude={151.25584}
      picture="images/tn_property01.jpg"
      price={937180}
    />
    <Card
      _id="593e9297ec4cca9c56bf61af"
      address="255 Raleigh Place"
      bathrooms={1}
      bedrooms={2}
      carSpaces={0}
      city="Machias"
      index={1}
      key="593e9297ec4cca9c56bf61af"
      latitude={-33.944471}
      longitude={151.2541}
      picture="images/tn_property02.jpg"
      price={703725}
    />
  </ul>
</div>
`;
